app/src/main/java/cm/aptoide/pt/home/apps/AppsManager.java

```diff
@@ -65,13 +71,21 @@ public AppsManager(UpdatesManager updatesManager, InstallManager installManager,
     this.context = context;
     this.downloadFactory = downloadFactory;
     this.moPubAdsManager = moPubAdsManager;
-    this.promotionsManager = promotionsManager;
     this.aptoideInstallManager = aptoideInstallManager;
+    this.updatesNotificationManager = updatesNotificationManager;
+    this.secureSharedPreferences = secureSharedPreferences;
   }
 
   public Observable<List<UpdateApp>> getUpdatesList() {
-    return Observable.combineLatest(getAllUpdatesList(), getUpdateDownloadsList(),
-        this::mergeUpdates);
+    if (SecurePreferences.isUpdatesFirstLoad(secureSharedPreferences)) {
+      return updatesManager.refreshUpdates()
+          .startWith(updateFirstLoadUpdatesSettings())
+          .andThen(startUpdatesNotification())
+          .andThen(getAllUpdatesList());
```

Comment:
in terms of performance, is the loading of the apps view ok? time?

Reply:
yes, i didn't notice that much of an issue, eventhough i have 37 updates on my xiaomi for example (and a lot of apps installed). It takes a bit longer than usual, but it is normal as it is making the request (but still under 2 seconds) and obviously with this solution we will alwyas be dependent on our services response times.